### ISSUE TITLE
TypeError: can't pickle weakref objects / EOFError: Ran out of input

### ISSUE THREAD

**Issue by oknchm** [setnicka/ulozto-downloader#56]
I keep getting these errors, I searched but I'm not really sure how to fix it. I'm very new to this so please be considerate. 

I'm using Python 3.7 on Windows 10. 

> File:           debian-9.6.0-amd64-netinst.iso
URL:            https://ulozto.cz/file/TKvQVDFBEhtL/debian-9-6-0-amd64-netinst-iso
Download type:  CAPTCHA protected download
Size / parts:   291.0MB => 15 x 19.4MB
[Part 0]        Waiting for CAPTCHA...
[Part 1]        Waiting for CAPTCHA...
[Part 2]        Waiting for CAPTCHA...
[Part 3]        Waiting for CAPTCHA...
[Part 4]        Waiting for CAPTCHA...
[Part 5]        Waiting for CAPTCHA...
[Part 6]        Waiting for CAPTCHA...
[Part 7]        Waiting for CAPTCHA...
[Part 8]        Waiting for CAPTCHA...
[Part 9]        Waiting for CAPTCHA...
[Part 10]       Waiting for CAPTCHA...
[Part 11]       Waiting for CAPTCHA...
[Part 12]       Waiting for CAPTCHA...
[Part 13]       Waiting for CAPTCHA...
[Part 14]       Waiting for CAPTCHA...Traceback (most recent call last):
  File "c:\users\oknchm\appdata\local\programs\python\python37\lib\runpy.py", line 193, in _run_module_as_main
[Progress]       0.00 MB (0.00 %)       avg. speed: 0.00 MB/s   curr. speed: 0.00 MB/s  remaining: 0:00:00
  File "c:\users\oknchm\appdata\local\programs\python\python37\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\OKNCHM\AppData\Local\Programs\Python\Python37\Scripts\ulozto-downloader.exe\__main__.py", line 7, in <module>
  File "c:\users\oknchm\appdata\local\programs\python\python37\lib\site-packages\uldlib\cmd.py", line 44, in run
    d.download(args.url, args.parts, args.output)
  File "c:\users\oknchm\appdata\local\programs\python\python37\lib\site-packages\uldlib\downloader.py", line 306, in download
    self.captcha_process.start()
  File "c:\users\oknchm\appdata\local\programs\python\python37\lib\multiprocessing\process.py", line 112, in start
    self._popen = self._Popen(self)
  File "c:\users\oknchm\appdata\local\programs\python\python37\lib\multiprocessing\context.py", line 223, in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
  File "c:\users\oknchm\appdata\local\programs\python\python37\lib\multiprocessing\context.py", line 322, in _Popen
    return Popen(process_obj)
  File "c:\users\oknchm\appdata\local\programs\python\python37\lib\multiprocessing\popen_spawn_win32.py", line 89, in __init__
    reduction.dump(process_obj, to_child)
  File "c:\users\oknchm\appdata\local\programs\python\python37\lib\multiprocessing\reduction.py", line 60, in dump
    ForkingPickler(file, protocol).dump(obj)
TypeError: can't pickle weakref objects
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "c:\users\oknchm\appdata\local\programs\python\python37\lib\multiprocessing\spawn.py", line 105, in spawn_main
    exitcode = _main(fd)
  File "c:\users\oknchm\appdata\local\programs\python\python37\lib\multiprocessing\spawn.py", line 115, in _main
    self = reduction.pickle.load(from_parent)
EOFError: Ran out of input

**Comment by arbarbosa1979** [comment]
I'm having the same kind of error, it doesn't work here at all, I tested in python 3.8 and 3.7 and the result is always the same.
```File:           O.V.1993.rar
URL:            https://ulozto.net/file/H1VewOgRi546/o-v-1993-rar#!ZGx4MQR2ZzWwMwR5BJRkMTL4LGSyAmL0AJyZIIueZTMPE2V3MD==
Download type:  CAPTCHA protected download
Size / parts:   2621.28MB => 15 x 187.23MB
[Part 0]        Waiting for CAPTCHA...
[Part 1]        Waiting for CAPTCHA...
[Part 2]        Waiting for CAPTCHA...
[Part 3]        Waiting for CAPTCHA...
[Part 4]        Waiting for CAPTCHA...
[Part 5]        Waiting for CAPTCHA...
[Part 6]        Waiting for CAPTCHA...
[Part 7]        Waiting for CAPTCHA...
[Part 8]        Waiting for CAPTCHA...
[Part 9]        Waiting for CAPTCHA...
[Part 10]       Waiting for CAPTCHA...
[Part 11]       Waiting for CAPTCHA...
[Part 12]       Waiting for CAPTCHA...
[Part 13]       Waiting for CAPTCHA...Traceback (most recent call last):
  File "ulozto-downloader.py", line 7, in <module>
    cmd.run()
  File "C:\Github\JudgeUAu\ulozto-downloader\uldlib\cmd.py", line 44, in run
    d.download(args.url, args.parts, args.output)
  File "C:\Github\JudgeUAu\ulozto-downloader\uldlib\downloader.py", line 306, in download
    self.captcha_process.start()
  File "c:\Python37\lib\multiprocessing\process.py", line 112, in start
    self._popen = self._Popen(self)
  File "c:\Python37\lib\multiprocessing\context.py", line 223, in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
  File "c:\Python37\lib\multiprocessing\context.py", line 322, in _Popen
Traceback (most recent call last):
    return Popen(process_obj)
  File "<string>", line 1, in <module>  File "c:\Python37\lib\multiprocessing\popen_spawn_win32.py", line 65, in __init__

    reduction.dump(process_obj, to_child)
  File "c:\Python37\lib\multiprocessing\spawn.py", line 105, in spawn_main  File "c:\Python37\lib\multiprocessing\reduction.py", line 60, in dump
    ForkingPickler(file, protocol).dump(obj)
TypeError: can't pickle weakref objects
    exitcode = _main(fd)
  File "c:\Python37\lib\multiprocessing\spawn.py", line 115, in _main
    self = reduction.pickle.load(from_parent)
EOFError: Ran out of input
```

**Comment by setnicka** [comment]
Hopefully solved by a switch from multiprocessing to multithreading (#93) and published as a version [3.0.0](https://pypi.org/project/ulozto-downloader/3.0.0/)
